pkg/compiler/transformers/k8s/node.go

```diff
@@ -221,6 +221,30 @@ func buildTasks(tasks []*core.CompiledTask, errs errors.CompileErrors) map[commo
 				errs.Collect(errors.NewValueCollisionError(taskID, "Id", taskID))
 			}
 
+			// We don't want Annotation data available at task runtime for performance.
+			if flyteTask.Template != nil &&
+				flyteTask.Template.Interface != nil {
+
+				if flyteTask.Template.Interface.Inputs != nil &&
+					flyteTask.Template.Interface.Inputs.Variables != nil {
```

Comment:
If it makes you feel any better you can make this:
```suggestion
				if flyteTask.Template.Interface.Inputs != nil {
```

Since `range nil` will yield 0 iterations and is an accepted value by go... but 🤷🏽